libnmstate/nm/plugin.py

```diff
@@ -0,0 +1,204 @@
+                + nm_ipv6.get_route_running(self.client)
+            ),
+            Route.CONFIG: (
+                nm_ipv4.get_route_config(self.client)
+                + nm_ipv6.get_route_config(self.client)
+            ),
+        }
+
+    def get_route_rules(self):
+        return {
+            RouteRule.CONFIG: (
+                nm_ipv4.get_routing_rule_config(self.client)
+                + nm_ipv6.get_routing_rule_config(self.client)
+            )
+        }
+
+    def get_dns_client_config(self):
+        return {
+            DNS.RUNNING: nm_dns.get_running(self.client),
+            DNS.CONFIG: nm_dns.get_config(
+                nm_ipv4.acs_and_ip_profiles(self.client),
+                nm_ipv6.acs_and_ip_profiles(self.client),
+            ),
+        }
+
+    def get_checkpoint(self):
+        return str(self._checkpoint.dbuspath) if self._checkpoint else None
+
+    def _load_checkpoint(self, checkpoint_path):
+        all_checkpoint_paths = get_checkpoints(self.client)
+        if not all_checkpoint_paths or (
+            checkpoint_path and checkpoint_path not in all_checkpoint_paths
+        ):
+            raise NmstateValueError(
+                f"Specified checkpoint {checkpoint_path} does not exists"
+            )
+        self._checkpoint = CheckPoint(
+            self._ctx,
+            dbuspath=checkpoint_path
+            if checkpoint_path
+            else all_checkpoint_paths[0],
+        )
+
+    def create_checkpoint(self, rollback_timeout):
+        if self._checkpoint:
+            raise NmstateConflictError("Only single checkpoint is allowed")
+        self._checkpoint = CheckPoint(self._ctx, timeout=rollback_timeout)
+        try:
+            self._checkpoint.create()
+        except NMCheckPointPermissionError:
+            raise NmstatePermissionError("Error creating a check point")
+        except NMCheckPointCreationError:
+            raise NmstateConflictError("Error creating a check point")
+        return self._checkpoint
+
+    def rollback_checkpoint(self, checkpoint_path=None):
+        self._load_checkpoint(checkpoint_path)
+        self._checkpoint.rollback()
+        self._checkpoint = None
+
+    def destroy_checkpoint(self, checkpoint_path=None):
+        self._load_checkpoint(checkpoint_path)
+        self._checkpoint.destroy()
+        self._checkpoint = None
+
+    def refresh_content(self):
+        self._ctx.refresh_content()
+
+
+def _ifaceinfo_bond(devinfo):
+    # TODO: What about unmanaged devices?
+    bondinfo = nm_translator.Nm2Api.get_bond_info(devinfo)
+    if "link-aggregation" in bondinfo:
+        return bondinfo
+    return {}
+
+
+def _remove_ovs_bridge_unsupported_entries(iface_info):
+    """
+    OVS bridges are not supporting several common interface key entries.
+    These entries are removed explicitly.
+    """
+    iface_info.pop(Interface.IPV4, None)
+    iface_info.pop(Interface.IPV6, None)
+    iface_info.pop(Interface.MTU, None)
+
+    return iface_info
```

Comment:
I do not understand why is this here. I mean, it is fine but how is this related to the NetworkManager plugin?

Reply:
This is copy of the original code from `netinfo.py`.
I just move the state query function into plugin realm.

Reply:
Sorry, but I could not understand why this belongs here. I would not block the PR because of that, but I will take a note and try to place it in a more proper place in the future. Thanks!

Reply:
It's https://github.com/nmstate/nmstate/blob/master/libnmstate/netinfo.py#L137 since that's plugin's job, I move it into plugin.

Reply:
Good, I will not block on that as in my opinion is a minor nit and probably not worth to discuss this. Thanks!